torch/csrc/jit/runtime/operator.h

```diff
@@ -40,51 +40,60 @@ using OperationCreator = Operation (*)(const Node*);
  * symbolic derivatives, which also requires them to have static lifetime
  * so that changes to symbolic derivatives are remembered.
  *
- * Now, currently, the c10 operator library doesn't store jit::Operator instances,
- * but we use a listener pattern that notifies JIT about changes in the
- * c10 operator library and then registers jit::Operator instances to the JIT
- * operator registry, acting as wrappers to the c10 operators.
- *
- * However, that results in code duplication as JIT and c10 will likely get
- * their own mechanisms for storing derivatives and other operator related
- * information, and all of this would have to be wrapped from c10 into JIT.
- *
- * We should consider merging the JIT and c10 registries, moving jit::Operator
- * to c10 and storing these jit::Operator instances in the c10 operator library
- * instead, allowing us to have these mechanisms only implemented once.
- * However, the current jit::Operator implementation has additional features
- * like OperationCreator that aren't needed in c10 (they're only used for
- * prim ops like If/Else or While which wouldn't be in the c10 operator library),
- * and which depend on other JIT features which we don't want to move to c10
- * (notably jit/ir.h). We might, however, be able, to split jit::Operator into
- * a c10::Operator with the core features and a jit::Operator that adds the
- * JIT-only features like OperationCreator, and then use c10::Operator in the
- * c10 operator library.
+ * Currently, the JIT operator library contains a jit::Operator instance
+ * with a wrapper for each c10 operator. The c10 operator library registers
+ * those wrappers using listeners in register_c10_ops.cpp.
+ * TODO Instead of doing it this way, we should only have pure-jit ops in
+ * the jit library but have the JIT operator lookup look into the c10 library too.
  */
 
+// An Operator is a thin wrapper around either a pure JIT operator (e.g. prim ops)
+// or a c10 operator, allowing some common operations and abstracting away the
+// concrete operator nature.
 struct TORCH_API Operator {
+private:
+  struct C10Operator final {
+    c10::OperatorHandle handle_;
+    Operation op_;
+  };
+  struct UnmaterializedFunctionSchema final {
```

Comment:
unmaterialized seems to be a new thing in this PR, so it would be helpful to briefly comment what this is. (It's a just a schema that hasn't been parsed yet, righT? Maybe UnparsedFunctionSchema is more direct.)